src/libcore/char.rs

```diff
@@ -420,17 +430,18 @@ pub fn encode_utf16_raw(mut ch: u32, dst: &mut [u16]) -> Option<usize> {
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct EscapeUnicode {
     c: char,
-    state: EscapeUnicodeState
+    state: EscapeUnicodeState,
+    offset: usize,
 }
 
 #[derive(Clone)]
 enum EscapeUnicodeState {
-    Backslash,
-    Type,
-    LeftBrace,
-    Value(usize),
-    RightBrace,
     Done,
+    RightBrace,
```

Comment:
This is just a micro-optimization (so that `state_len` is a single load instead of load+add).
I can make it more explicit and/or reference the comment in `state_len` or I can remove it if it is deemed undesirable.